### DISCUSSION TITLE
GeoJson layer: non-visible feature due to z coordinate in orthographic view

### DISCUSSION THREAD

**Discussion by hoang-ta** [visgl/deck.gl#5974] [Q&A] [answered]
Hello everyone,  currently i am displaying GeoJson features with z coordinate in the Orthographic view. My initial view state target is **[0,0,800]**. You can view my code demo at this  [sandbox url](https://codesandbox.io/s/awesome-spence-8qr5q?file=/src/App.tsx)

- The problem is: 
1. Only features whose z values that are smaller than initial view state z target, are visible (green points):
![highest-z-2](https://user-images.githubusercontent.com/61722520/125759925-eb1e43d3-2bda-4768-b20b-a887ab0fa9a5.GIF)

2. If I set the initial view state z target to be the highest, **green points** whose z values are low will **disappear on zoom:**
![highest-z-disappear-on-zoom](https://user-images.githubusercontent.com/61722520/125759949-c459c731-6639-4835-a30c-a484940df229.GIF)

- What i am trying to solve: 

1.Display features with z coordinate (I can do this by setting high value for the initial view state target z). I am stuck at number 2
2. Prevent features disappear on zoom 

Thank you for your time.

**Comment by Pessimistress** [comment, marked as answer]
You can try change the `near` and `far` parameters of the OrthographicView.

Ultimately, the z does not matter in orthographic view, and you can flatten your data like so

```js
import {Matrix4} from 'math.gl';

new GeoJsonLayer({
  ...
  modelMatrix: new Matrix4().scale([1,1,0])
});
```

**Reply by hoang-ta** [reply]
Thank you, although i'm not sure what really makes it work :)

- I test with `near` parameter,  set its value to **negative**, decrease it, and it works without setting high value for the initial view state target z (now i can use default [0,0,0])
- As I notice, the lower the `near` value, the more zoom in I can perform until the object  with high z value (red point) disappears. 
- I wonder if there is any document about this topic.
